### PR TITLE
Little fixes

### PR DESCRIPTION
**Pull request by ChrisJefferson**
This is a collection of tiny fixes, which mostly deal with crashes when giving functions silly arguments.

The final commit makes TNUMs unsigned, to line up with the original macro-based definitions.

### CODE REVIEW

File: src/funcs.c [hunk 1]
```diff
@@ -1909,12 +1909,12 @@ void            ExecEnd (
 *F  FuncSetRecursionTrapInterval( <self>, <interval> )
 **
 */
-
+#include <stdio.h>
```
Comment (fingolfin) [inline src/funcs.c]: Any reason why this #include is down here, and not at the top of the file with the others? For that matter: Why did you add it at all?

Comment (fingolfin) [inline src/funcs.c]: BTW, the first line of the commit message "Ensure RecursionTrapInterval is big enough we can enter the break loop without hitting it" exceeds 70 chars and gets wrapped by GitHub. At the same time, I did not really understand it, and had to look at the content of the commit.

How about this:
```
Ensure RecursionTrapInterval is always at least 5

For smaller values, using the break loop becomes impossible
```

File: src/plist.c [hunk 2]
```diff
@@ -915,6 +915,9 @@ Obj    FuncEmptyPlist( Obj self, Obj len )
             "you can replace <len> via 'return <len>;'" );
     }
 
+    if(INT_INTOBJ(len) < 0) {
```
Comment (fingolfin) [inline src/plist.c]: Space after `if` is missing

File: src/plist.c [hunk 3]
```diff
@@ -915,6 +915,9 @@ Obj    FuncEmptyPlist( Obj self, Obj len )
...
+        ErrorMayQuit("<len> must be greater than or equal 0",0L,0L);
```
Comment (fingolfin) [inline src/plist.c]: how about "must be non-negative" or  "must be a non-negative small integer" as we use elsewhere?

File: src/blister.c [hunk 4]
```diff
@@ -1568,7 +1568,7 @@ Obj FuncLIST_BLIST (
     while ( LEN_LIST( list ) != LEN_BLIST( blist ) ) {
         blist = ErrorReturnObj(
             "ListBlist: <blist> must have the same length as <list> (%d)",
-            LEN_PLIST( list ), 0L,
+            LEN_LIST( list ), 0L,
```
Comment (fingolfin) [inline src/blister.c]: I find the commit message unclear. How about "Stop calling LEN_PLIST on generic lists" or "Avoid ..." -- when you say "fix", that could also mean that you are changing things so that this actually becomes valid.

File: src/permutat.c [hunk 5]
```diff
@@ -3357,12 +3357,12 @@ Obj             FuncRESTRICTED_PERM (
 	ptDom  = ADDR_OBJ(dom);
 	len = LEN_LIST(dom);
 	for (i=1;i<=len;i++) {
-	  p=INT_INTOBJ(ptDom[i]);
-	  if ( IS_INTOBJ(ptDom[i]) && p>0 ) {
-	    if (p<=deg) {
-	      p-=1;
-	      ptRest2[p]=ptPerm2[p];
-	    }
+            if (IS_INTOBJ(ptDom[i]) && INT_INTOBJ(ptDom[i]) > 0) {
```
Comment (fingolfin) [inline src/permutat.c]: You could also us `IS_POS_INTOBJ` here

File: src/permutat.c [hunk 6]
```diff
@@ -3431,17 +3431,17 @@ Obj             FuncRESTRICTED_PERM (
...
-	    if (p<=deg) {
-	      p-=1;
-	      ptRest4[p]=ptPerm4[p];
-	    }
-	  }
-	  else{
-	    return Fail;
-	  }
+        for (i = 1; i <= len; i++) {
+            if (IS_INTOBJ(ptDom[i]) && INT_INTOBJ(ptDom[i]) > 0) {
```
Comment (fingolfin) [inline src/permutat.c]: Again, could us `IS_POS_INTOBJ`

File: src/blister.h [hunk 7]
```diff
@@ -52,7 +52,7 @@ static inline Int IS_BLIST_REP(Obj list)
 
 static inline Int IS_BLIST_REP_WITH_COPYING(Obj list)
 {
-    Int tnum = TNUM_OBJ(list);
+    UInt tnum = TNUM_OBJ(list);
```
Comment (fingolfin) [inline src/blister.h]: Commit message is "Treat TNUMs are UInts, not Ints" -> "are" should b "as", I guess?

File: src/funcs.c (hunk 1 above)
Comment (ChrisJefferson) [inline src/funcs.c]: That header is just a debugging thing I left in, sorry.

Comment (fingolfin) [inline src/funcs.c]: That's why we do code review, no need to say "sorry" (I did far worse things in the past ... let's not talk about it ;-)

File: src/gap.c [hunk 8]
```diff
@@ -2017,17 +2017,28 @@ Obj FuncHANDLE_OBJ (
     return hnum;
 }
 
-/* This function does quite  a similar job to HANDLE_OBJ, but (a) returns 0 for all 
-immediate objects (small integers or ffes) and (b) returns reasonably small results
-(roughly in teh range from 1 to the max number of objects that have existed in this session */
+/* This function does quite  a similar job to HANDLE_OBJ, but (a) returns 0
+for all immediate objects (small integers or ffes) and (b) returns reasonably
+small results (roughly in teh range from 1 to the max number of objects that
```
Comment (fingolfin) [inline src/gap.c]: If hou reformat this anyway, might as well fix "teh" typo

File: src/plist.c [hunk 9]
```diff
@@ -915,6 +915,9 @@ Obj    FuncEmptyPlist( Obj self, Obj len )
             "you can replace <len> via 'return <len>;'" );
     }
 
+    if (INT_INTOBJ(len) < 0) {
+        ErrorMayQuit("<len> must be a non-negative small integer", 0L, 0L);
+    }
```
Comment (fingolfin) [inline src/plist.c]: It seems odd that one validation check allows me to enter a new value, the other doesn't. Why not just merge the two into a single one, anyway? I.e. `while ( ! intobj || value < 0)` ?

File: tst/testbugfix/2017-06-20-blist-list-len.tst [hunk 10]
```diff
@@ -0,0 +1,17 @@
+# Actual bug -- checking lists are the same length
```
Comment (fingolfin) [inline tst/testbugfix/2017-06-20-blist-list-len.tst]: What does "Actual bug" mean here exactly? Doesn't seem to convey any useful/helpful information :-(

Comment (fingolfin) [inline tst/testbugfix/2017-06-20-blist-list-len.tst]: Also, now about instead of a bugfix file, use this as start for a `blist.tst`? I.e. Just move&rename the file

File: tst/testbugfix/2017-06-20-emptyplist.tst [hunk 11]
```diff
@@ -0,0 +1,16 @@
+gap> EmptyPlist(-1);
```
Comment (fingolfin) [inline tst/testbugfix/2017-06-20-emptyplist.tst]: This file could also be `opers/EmptyPlist.tst`, or part of a new `plist.tst`
